src/object/index.spec.ts

```diff
@@ -174,13 +177,15 @@ describe("object", () => {
     }));
 
   it("mocks the same value with the same path", () => {
-    const objectDef = () => ({
+    const objectDef = (): {
+      fields: TupleOfLength<FieldOptions<"foo", ZodAny, unknown, boolean>, 1>;
```

Comment:
ah, this is what I wanted to avoid. I want to infer the types rather than declare them here, which is challenging without the `as const`.

Reply:
Maybe there's a solution we use `as const`, then translate it's type to not be `readonly`? Or maybe something along the lines of:

```typescript
const field = {
          name: "foo",
          type: string(),
        };

// Might not even need the type declaration here, try it out.
const fields: [typeof field] = [field];

return ({ fields });
```

Reply:
That doesn't work unfortunately.

This does work, but is still a little awkward.
```typescript
    const objectDef = () => ({
      fields: [
        {
          name: "foo",
          type: string(),
        },
      ],
    });
    

    expect(object(objectDef() as DeepWriteable<ReturnType<typeof objectDef>>)
```

Got any better ideas?

Reply:
Actually no, that doesn't work either.. :/

Reply:
When you say my suggestion doesn't work, what's the error you're getting exactly?

Reply:
@saiichihashimoto Sorry, i must've made a typo somewhere, got distracted multiple times while trying to figure out a fix, it does work!

```typescript
    const objectDef = () => {
      const field = {
        name: "foo",
        type: string(),
      };

      const fields: [typeof field] = [field];

      return ({ fields });
    };
 ```
 It does need the type definition though, but that's muuuch cleaner.

Reply:
That looks great

Reply:
fixed :+1: